src/main/java/com/bravo/user/dao/specification/AbstractSpecification.java

```diff
@@ -79,28 +80,57 @@ protected void applyDateTimeFilter(
     final List<String> inClause = new ArrayList<>();
 
     for(String s : values){
-      final boolean isLike = s.contains("%") || s.contains("*");
-      final boolean isNot = s.startsWith("!");
+      final boolean isLike = isLike(s);
+      final boolean isNot = isNot(s);
 
       if(!isLike && !isNot){
         inClause.add(s.toLowerCase());
         continue;
       }
       final String targetValue = (isNot ? s.substring(1) : s).toLowerCase();
-      Predicate predicate;
-      if(isLike){
-        predicate = criteriaBuilder.like(targetPath, targetValue.replace("*", "%"));
-      } else {
-        predicate = criteriaBuilder.equal(targetPath, targetValue);
-      }
-
-      if(isNot){
-        predicate = criteriaBuilder.not(predicate);
-      }
+      Predicate predicate = createPredicate(targetPath, targetValue, isLike, isNot);
       predicates.add(predicate);
     }
     if(!inClause.isEmpty()){
       applyInFilter(targetPath, inClause);
     }
   }
+
+  protected void applyStringFilterToFields(Collection<Expression<String>> fields, String value) {
```

Comment:
Is it possible to overload the `applyStringFilter` method instead of duplicating some of the code used in that method?

Reply:
It is possible, but I'm not sure it would be cleaner, the extraction of `createPredicate` eliminated most of the duplicated code.  It would be good to breifly pair with you to get a better understanding of what you are thinking.